lib/_tls_wrap.js

```diff
@@ -1085,12 +1085,34 @@ exports.connect = function(/* [port, host], options, cb */) {
     socket.removeListener('end', onHangUp);
   });
 
+  var socketAddress;
+
+  if (options.socket) {
+    options.socket.once('connect', function() {
+      socketAddress = {
+        host: options.socket.remoteAddress,
+        port: options.socket.remotePort
+      };
+    });
+  } else {
+    if (options.path && !options.port) {
+      socketAddress = { path: options.path };
+    } else {
+      socketAddress = {
+        port: options.port,
+        host: options.host,
+        localAddress: options.localAddress
+      };
+    }
+  }
+
```

Comment:
@bnoordhuis ok, I am going to revert to what I had before, but I won't be able to handle the `options.socket` case properly, because it seems that `socket.remoteAddress` is deleted before the `end` event.